xbmc/platform/freebsd/CPUInfoFreebsd.cpp

```diff
@@ -135,3 +239,20 @@ float CCPUInfoFreebsd::GetCPUFrequency()
 
   return static_cast<float>(hz);
 }
+
+
+bool CCPUInfoFreebsd::GetTemperature(CTemperature& temperature)
+{
+  int value;
+  size_t len;
+
+  len = sizeof(value);
+  if (sysctlbyname("dev.cpu.0.temperature", &value, &len, NULL, 0) != 0)
+    return false;
```

Comment:
you should do what we do for linux here https://github.com/xbmc/xbmc/blob/master/xbmc/platform/linux/CPUInfoLinux.cpp#L299

This will allow the test to be run regardless of the hardware.